vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs

```diff
@@ -67,27 +67,23 @@ type internal FSharpDocumentDiagnosticAnalyzer [<ImportingConstructor>] () =
             use _eventDuration =
                 TelemetryReporter.ReportSingleEventWithDuration(TelemetryEvents.GetDiagnosticsForDocument, eventProps)
 
+
             let! ct = CancellableTask.getCurrentCancellationToken ()
 
             let! parseResults = document.GetFSharpParseResultsAsync("GetDiagnostics")
 
             let! sourceText = document.GetTextAsync(ct)
             let filePath = document.FilePath
 
-            let! errors =
-                cancellableTask {
-                    match diagnosticType with
-                    | DiagnosticsType.Semantic ->
-                        let! _, checkResults = document.GetFSharpParseAndCheckResultsAsync("GetDiagnostics")
-                        // In order to eliminate duplicates, we should not return parse errors here because they are returned by `AnalyzeSyntaxAsync` method.
-                        let allDiagnostics = HashSet(checkResults.Diagnostics, diagnosticEqualityComparer)
-                        allDiagnostics.ExceptWith(parseResults.Diagnostics)
-                        return Seq.toArray allDiagnostics
-                    | DiagnosticsType.Syntax -> return parseResults.Diagnostics
-                }
+            let errors = HashSet<FSharpDiagnostic>(parseResults.Diagnostics, diagnosticEqualityComparer)
```

Comment:
One more thought -this gets called very often and a typical case will be about not having any errors alltogether.

Maybe the parseResults + checkResults could be collected first, and a non-allocating codepath taken in case there are no errors whatsover?